### DISCUSSION TITLE
Change form output from a string to an array

### DISCUSSION THREAD

**Discussion by smack322** [react-hook-form/react-hook-form#3415] [Q&A]
I currently have a react form that was created with the custom hook: react-hook-form - https://react-hook-form.com/. I have an input box that I would like to output an array of comma separated values. i.e. "test1", "test2". The current implementation that I have outputs the data as "test1, test2". Is anyone familiar with a way to get the value back as an array and not a string? Here is what I have.
```
    <label className="form-label" htmlFor="[tenant_authorized_guid]">Tenant_Authorized_Guid(s): 
    <input
      name="[tenant_authorized_guid]"
      placeholder="guid1, guid2, guid3"
      ref={register({
        required: "Required",
        validate: (value) => value !== "",
      })}
    />
```
I am thinking I need to use the controller / useFieldArray but am pretty new to this.

**Comment by bluebill1049** [comment]
is this what you after? https://react-hook-form.com/advanced-usage#TransformandParse

**Reply by smack322** [reply]
Thanks for the quick reply. It's close. Just one input box that takes in more than one value and spits out an array of comma separated values. The data type would be strings inside of the array.

input: test1, test2, test3
output: ['test1', 'test2', 'test3'].

Some way to use .split(',') against the value(s) from the input box

**Reply by bluebill1049** [reply]
you couldn't use 
```
output: (e) => {
      const output = parseInt(e.target.value, 10);
      return isNaN(output) ? 0 : output;
    }```

do similar .split?

**Reply by smack322** [reply]
Will the transform and parse work with a string? I can play around with trying to get the output to split.
